src/models/user-data.ts

```diff
@@ -0,0 +1,36 @@
+import { SetDocument } from "./set-document";
+
+
+/** 
+ * This model is for mapping user information retrieved from /users and persisting it to the user's profile.
+ * @author Cody McDonald
+ * 
+*/
+
+export class UserData {
+    id: string;
+    username: string;
+    favoriteSets: Array<SetDocument>;
+    createdSets: Array<SetDocument>;
+    profilePicture: string;
+    points: number;
+    wins: number;
+    losses: number;
+    registrationDate: string;
+    gameRecords: Array<string>;
+
+    constructor(id?: string, username?: string, favoriteSets?: Array<SetDocument>, createdSets?: Array<SetDocument>,
```

Comment:
There has to be a way to indent this better lmao

Reply:
Tried something else, probably will still look ugly since there's so much in the model.